README.md

```diff
@@ -43,83 +43,123 @@ tarpc-plugins = { git = "https://github.com/google/tarpc" }
-The `service!` macro expands to a collection of items that form an
-rpc service. In the above example, the macro is called within the
-`hello_service` module. This module will contain `SyncClient`, `AsyncClient`,
-and `FutureClient` types, and `SyncService` and `AsyncService` traits.  There is
-also a `ServiceExt` trait that provides starter `fn`s for services, with an
-umbrella impl for all services.  These generated types make it easy and
-ergonomic to write servers without dealing with sockets or serialization
-directly. Simply implement one of the generated traits, and you're off to the
-races! See the `tarpc_examples` package for more examples.
+In this example we call `listen("localhost:0", ..)` to bind the `HelloServer` object to a port chosen by
+the operating system. We then print the chosen port to stdout so a client can know which port to
+connect to. In a production service it's more common to specify the port as a commandline argument.
+
+Next is the command line client. This client allows you to specify a server address and a person's
+name, and sends a `hello` RPC with that person's name to the specified server. For simplicity this
+binary uses `clap` to handle command line arguments, though it's not required.
+
+```rust
+extern crate clap;
+extern crate tarpc;
+extern crate sync_starter;
+
+use tarpc::client::Options;
+use tarpc::client::sync::ClientExt;
+
+fn main() {
+    let matches = clap::App::new("hello sync client")
+        .arg(clap::Arg::with_name("server_address").required(true))
+        .arg(clap::Arg::with_name("person_name").required(true))
+        .get_matches();
+    let addr = matches.value_of("server_address").unwrap();
+    let person_name = matches.value_of("person_name").unwrap();
+    let mut client = sync_starter::SyncClient::connect(addr, Options::default()).unwrap();
+    println!("{}", client.hello(person_name.into()).unwrap());
+}
+```
+
+Code for this example can be found in `examples/sync_starter/src`. To run it do the following:
+
+```
+$ cd examples/sync_starter
+$ cargo build
+$ ../../target/debug/sync_server
+Listening on [::1]:61081
+```
+
+Now the server is listening on port `61081`. We can then use the generated client to send an RPC to
+the server:
+
+```
+$ ../../target/debug/sync_client "[::1]:61081" Mom
+Hey Mom!
+```
 
 ## Example: Futures
 
-Here's the same service, implemented using futures.
+Here's the same service, implemented using futures. The main difference is that we now return a
+future from service implementation methods.
 
 ```rust
 #![feature(plugin)]
 #![plugin(tarpc_plugins)]
 
-extern crate futures;
 #[macro_use]
 extern crate tarpc;
-extern crate tokio_core;
 
-use futures::Future;
-use tarpc::future::{client, server};
-use tarpc::future::client::ClientExt;
-use tarpc::util::{FirstSocketAddr, Never};
-use tokio_core::reactor;
+use tarpc::util::Never;
 
 service! {
     rpc hello(name: String) -> String;
 }
 
 #[derive(Clone)]
-struct HelloServer;
+pub struct HelloServer;
```

Comment:
Same comment as above about having this code in a different module than the service declaration, to encourage good habits.